charts/sonarqube/values.schema.json

```diff
@@ -205,5 +200,68 @@
                 }
             }
         }
-    }
- }
+    },
+    "allOf": [
```

Comment:
Based on the acceptance criteria of the task, I think that adding this strong validation of the fields is not required.

Specifically, when edition is set, as of now we always favor this, regardless of whether community.enabled is set or not. The downside of adding the strong validation is to have quite a few warnings (not really meaningful) written when edition is not set. 

E.g.
```
Error: values don't meet the specifications of the schema(s) in the following chart(s):
sonarqube:
- (root): Must validate "then" as "if" was valid
- (root): edition is required
- (root): Must validate all the schemas (allOf)
```
Only the second warning is really actionable. 

Having said that, shall we simplify the validation? wdyt?